### DISCUSSION TITLE
How calculate the distance and time between two drivers at a specific time?

### DISCUSSION THREAD

**Discussion by nicZox** [theOehrly/Fast-F1#206] [Q&A]
Hello,
How can I calculate with Fast-F1 the distance and time between two drivers at a specific time?
I would like to do something similar to this graphic but I can't figure out how to derive the distance and time at 15:57.11 (when the safety car is deployed) between Sainz and Leclerc.

Can you give me some help?
Thank you

![image](https://user-images.githubusercontent.com/76594617/177381487-7ca86e14-fc36-423a-a369-a0d3af79030b.png)
![image](https://user-images.githubusercontent.com/76594617/177381500-b1f189fa-9a7c-458f-a441-3fa8842b0372.png)

**Comment by theOehrly** [comment]
Well, this is not really a FastF1 problem, more like a kinematics problem I'd say 😁

Distance: Take the current lap for two drivers. Get telemetry with distance data. Get the distance for both drivers at the time when the safety car is deployed. Subtract one from the other.

Time: The question to be answered is how long ago was the first driver at the point where the second driver is now. 
Take the current lap for two drivers. Get telemetry with distance data. Get the distance at which the second driver is now. Find the timestamp at which the first driver already was at this distance. This was x seconds before now. This x is your delta time between the two drivers 

Hope this helps
